### PR TITLE
Remove Logger.OSThread-based PID logging that was always disabled

### PR DESCRIPTION
**Pull request by bertm**
The PID logging is always disabled and there is no API for enabling it. Remove the entire class and all the calls to its logPID method.

### CODE REVIEW

File: src/freenet/support/Logger.java [hunk 1]
```diff
@@ -1,206 +1,17 @@
...
 import freenet.support.FileLoggerHook.IntervalParseException;
 import freenet.support.LoggerHook.InvalidThresholdException;
-import freenet.support.io.Closer;
 
 /**
  * @author Iakin
 
 */
 public abstract class Logger {
-	public final static class OSThread {
```
Comment (ArneBab) [inline src/freenet/support/Logger.java:25]: used in SCGIPublisher ⇒ can’t just remove.

Comment (ArneBab) [reply src/freenet/support/Logger.java:25]: ⇒ we need a shim (that doesn’t do anything).

Comment (bertm) [reply src/freenet/support/Logger.java:25]: No sane plugin^W^W^W Restored the Logger$OSThread methods, but removed their implementation.
Thanks for catching!

Comment (ArneBab) [reply src/freenet/support/Logger.java:25]: Thank you! SCGIPublisher is actually sane and useful (hosting hyphanet inproxies with strict allow-list), but it is also old and unmaintained.

**Review by ArneBab — APPROVED** [review]

**Comment by ArneBab** [comment]
merged — thank you!

**Pull request merged**